File: inventory-ui-pro/src/services/userService.js
import API from "./api";

export const loginUser = async (credentials) => {
  const res = await API.post("/users/login", credentials);
  return res.data;
};

export const registerUser = async (user) => {
  const res = await API.post("/users/register", user);
  return res.data;
};

export const getAllUsers = async () => {
  const res = await API.get("/users");
  return res.data;
};

export const getUserByUsername = async (username) => {
  const res = await API.get(`/users/${username}`);
  return res.data;
};

File: inventory_forecasting_backend/src/main/java/com/inventory/service/UserService.java
package com.inventory.service;

import java.util.Optional;
import com.inventory.entity.User;

public interface UserService {

    // REGISTER
    User saveUser(User user);

    // LOGIN
    Optional<User> login(String username, String password);
    Optional<User> getUserByUsername(String username);
    java.util.List<User> getAllUsers();

}
